### PR TITLE
Change cache to Read-Through Write-Behind cache

### PR DESCRIPTION
**Pull request by varungu**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @varungu has signed the CLA. Thanks for the pull request!

File: box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java [hunk 1]
```diff
@@ -24,8 +22,7 @@
 
     private static final long serialVersionUID = 7174936367401884790L;
     // Map that holds all the properties of the entity. LinkedHashMap was chosen to preserve ordering when outputting json
-    protected JsonObject mJsonObject;
-    transient CacheMap mCacheMap;
+    CacheMap mCacheMap;
```
Comment (doncung) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java]: If this is not transient we need to change the private writ and read logic. Also not sure if we want to even bother restoring cached values.

**Comment by doncung** [comment]
I am not opposed to adding setters, but I want to get the rest of the teams thoughts.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java [hunk 2]
```diff
@@ -214,6 +276,13 @@ public String getAsString(final String field){
             return value.asString();
         }
 
+        public void set(final String field, final String value){
+            mJsonObject.add(field, value);
```
Comment (doncung) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java]: There is a difference between add and set here so we need to be careful.

Comment (codepathreview) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java]: Sure. We can take a look and use the correct ones.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java (hunk 1 above)
Comment (codepathreview) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxJsonObject.java]: We need to move the same read and write logic to cache. There is no need to restore the map but we need to initialize the same in read.

**Pull request merged**